test/Test/Database/LSMTree/ModelIO/Monoidal.hs

```diff
@@ -178,13 +181,54 @@ prop_insertLookupRange h ups k v r = ioProperty $ do
 -- implement classic QC tests for monoidal updates
 -------------------------------------------------------------------------------
 
-{- TODO -}
+-- | You can lookup what you inserted.
+prop_lookupUpdate ::
+     IsTableHandle h
+  => Proxy h -> [(Key, Update Value)]
+  -> Key  -> Value -> Value -> Property
+prop_lookupUpdate h ups k v1 v2 = ioProperty $ do
+    -- create session, table handle, and populate it with some data.
+    (_, hdl) <- makeNewTable h ups
+
+    -- the main dish
+    inserts hdl [(k, v1)]
+    mupserts hdl [(k, v2)]
+    res <- lookups hdl [k]
+
+    -- notice the order.
+    return $ res === [Found k $ merge v2 v1]
 
 -------------------------------------------------------------------------------
 -- implement classic QC tests for monoidal table merges
 -------------------------------------------------------------------------------
 
-{- TODO -}
+-- this property doesn't work.
```

Comment:
hmm? in what sense?

Reply:
Ah, that is outdated comment. I'll remove it. (One of previous versions didn't work).